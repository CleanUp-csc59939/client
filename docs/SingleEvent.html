<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: SingleEvent</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: SingleEvent</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>SingleEvent<span class="signature">()</span><span class="type-signature"> &rarr; {Component}</span></h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    

    
    <h4 class="name" id="SingleEvent"><span class="type-signature"></span>new SingleEvent<span class="signature">()</span><span class="type-signature"> &rarr; {Component}</span></h4>
    

    



<div class="description">
    <p>returns page with single event, members that have joined and options for the event*</p>
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_meetings_event.jsx.html">components/meetings/event.jsx</a>, <a href="components_meetings_event.jsx.html#line39">line 39</a>
    </li></ul></dd>
    

    

    

    
</dl>















<h5>Returns:</h5>

        
<div class="param-desc">
    <p>Return single event component</p>
</div>



<dl>
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">Component</span>


    </dd>
</dl>

    





    
    </div>

    

    

    

    

    

    

    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    
    <h4 class="name" id="~renderEventActions"><span class="type-signature">(inner) </span>renderEventActions<span class="signature">()</span><span class="type-signature"> &rarr; {HTMLBodyElement}</span></h4>
    

    



<div class="description">
    <p>Renders buttons based on different user roles</p>
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_meetings_event.jsx.html">components/meetings/event.jsx</a>, <a href="components_meetings_event.jsx.html#line66">line 66</a>
    </li></ul></dd>
    

    

    

    
</dl>















<h5>Returns:</h5>

        
<div class="param-desc">
    <p>Returns the Join/Leave button if not owner of event, Return Edit and Delete event if the owner</p>
</div>



<dl>
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">HTMLBodyElement</span>


    </dd>
</dl>

    





        
    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="Banner.html">Banner</a></li><li><a href="CreateMeetUp.html">CreateMeetUp</a></li><li><a href="Header-CreateEvent.html">CreateEvent</a></li><li><a href="Home.html">Home</a></li><li><a href="Login.html">Login</a></li><li><a href="SingleEvent.html">SingleEvent</a></li></ul><h3>Global</h3><ul><li><a href="global.html#authHeader">authHeader</a></li><li><a href="global.html#ConvertDate">ConvertDate</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc/jsdoc">JSDoc 3.6.7</a> on Tue May 25 2021 00:33:43 GMT-0400 (Eastern Daylight Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>