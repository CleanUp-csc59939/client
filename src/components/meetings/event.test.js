import React from 'react';
import { mount } from 'enzyme';
import SingleEvent from './event';

const testUser = {
  userID: 15,
  name: null,
  bio: null,
  location: null,
  img: 'https://res.cloudinary.com/cleanup/image/upload/v1621121500/download_remvsp.png',
};

describe('SingleEvent', () => {
  it('should render correctly in "debug" mode', async () => {
    // const preventDefault = jest.fn()

    const match = { params: { id: 34 } };
    const component = mount(<SingleEvent match={match} currentUser={testUser} debug />);
    // expect(preventDefault).toHaveBeenCalledTimes(1)

    //

    // expect(component.containsMatchingElement(<div class="banner-subheader">The Test Event</div>)).toBeTruthy();
    expect(component).toMatchSnapshot();
  });
});
