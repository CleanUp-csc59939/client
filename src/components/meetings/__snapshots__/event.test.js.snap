// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleEvent should render correctly in "debug" mode 1`] = `
<SingleEvent
  currentUser={
    Object {
      "bio": null,
      "img": "https://res.cloudinary.com/cleanup/image/upload/v1621121500/download_remvsp.png",
      "location": null,
      "name": null,
      "userID": 15,
    }
  }
  debug={true}
  match={
    Object {
      "params": Object {
        "id": 34,
      },
    }
  }
>
  <div
    id="loading"
  >
    <img
      alt="Loading..."
      id="loading-image"
      src="loading/loading-gif.gif"
    />
  </div>
</SingleEvent>
`;
